api/processor_api.py

```diff
@@ -274,6 +274,50 @@ def handle_config_response(config, success):
                 )
             return JSONResponse(content=humps.decamelize(config))
 
+        def get_areas():
+            config = extract_config()
+            areas_name = [x for x in config.keys() if x.startswith("Area")]
+            return [map_area(x, config) for x in areas_name]
+
+        def reestructure_areas(config_dict):
+            """Ensure that all [Area_0, Area_1, ...] are consecutive"""
+            areas_names = [x for x in config_dict.keys() if x.startswith("Area")]
+            for index, area_name in enumerate(areas_names):
+                if f'Area_{index}' != area_name:
+                    config_dict[f'Area_{index}'] = config_dict[area_name]
+                    config_dict.pop(area_name)
+            return config_dict
+
+        def reestructure_cameras(config_dict):
+            """Ensure that all [Source_0, Source_1, ...] are consecutive"""
+            source_names = [x for x in config_dict.keys() if x.startswith("Source")]
+            for index, source_name in enumerate(source_names):
+                if f'Source_{index}' != source_name:
+                    config_dict[f'Source_{index}'] = config_dict[source_name]
+                    config_dict.pop(source_name)
+            return config_dict
+
+        def delete_camera_from_areas(camera_id, config_dict):
+            areas_names = [x for x in config_dict.keys() if x.startswith("Area")]
+            areas = [map_area(x, config_dict) for x in areas_names]
+            for index, area in enumerate(areas):
+                cameras = area['cameras'].split(',')
+                if camera_id in cameras:
+                    cameras.remove(camera_id)
+                    if len(cameras) == 0:
+                        config_dict.pop(f'Area_{index}')
```

Comment:
Instead of hardcoding the index, I think you should have the section name for this area so that you don't need the index

Reply:
How would you retrieve it? I've thought about adding it to each .ini, but doesn't seem like the best idea

Reply:
If you use `config_dict.items()` instead of keys, you get tuples of key and value. Then you probably don't need the next map_area.
And you have the area_{index} as key in the tuple